src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+                                                            final ArrayList<String> indexerArgs){
+
+        final SGAModule module = new SGAModule(moduleName);
+        final ArrayList<String> args = new ArrayList<>();
+        args.add("--threads");   args.add(Integer.toString(threads));
+        args.add(inputFASTAFile.getName());
+        final Tuple3<Integer, String, String> moduleInfo = module.run(sgaPath, outputDirectory, args);
+
+        final String outputFileName = extractBaseNameWithoutExtension(inputFASTAFile) + extensionToAppend;
+
+        final SGAUnitRuntimeInfoStruct unitInfo = new SGAUnitRuntimeInfoStruct(moduleName, moduleInfo, outputFileName);
+
+        indexerArgs.set(indexerArgs.size()-1, outputFileName);
+        final Tuple3<Integer, String, String> indexerInfo = indexer.run(sgaPath, outputDirectory, indexerArgs);
+        unitInfo.add("index", indexerInfo);
+
+        return unitInfo;
+    }
+
+    // From https://www.stackoverflow.com/questions/4545937
+    @VisibleForTesting
+    static String extractBaseNameWithoutExtension(final File file){
+        final String[] tokens = file.getName().split("\\.(?=[^\\.]+$)");
+        return tokens[0];
+    }
+
+    /**
+     * A struct to represent the runtime information returned by running a unit of sga modules.
+     * Because some of the modules are designed to be run together in serial form in a unit (e.g. correction followed by indexing)
+     *   all return status and stdout stderr messages in this unit are logged.
+     * But only one file name is important for consumption by downstream units/modules, so only that file name is kept.
+     * The ctor and the add() method are designed asymmetrically, because when a unit is consisted of multiple modules,
+     *   only one generates the file name to be returned, and the other modules are simply to be logged.
+     */
+    @VisibleForTesting
+    static final class SGAUnitRuntimeInfoStruct implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        public final String fileNameToReturn;
+        public final List<Tuple2<String, Tuple3<Integer, String, String>>> runtimeInfo;
+
+        public SGAUnitRuntimeInfoStruct(final String moduleName, final Tuple3<Integer, String, String> entry, final String fileName){
+            fileNameToReturn = fileName;
+            runtimeInfo = new ArrayList<>();
+            add(moduleName, entry);
+        }
+
+        public void add(final String moduleName, final Tuple3<Integer, String, String> entry){
+            runtimeInfo.add(new Tuple2<>(moduleName, entry));
+        }
+    }
+
+    /**
+     * Represents a collection of assembled contigs in the final output of "sga assemble".
+     */
+    @VisibleForTesting
+    static final class ContigsCollection implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        @VisibleForTesting
+        static final class ContigSequence implements Serializable{
```

Comment:
What's the reason for putting ID and Sequence into their own separate classes? It's fine to leave it this way, I'm just asking.